src/Middleware/HttpLogging/src/HttpLoggingExtensions.cs

```diff
@@ -29,6 +29,9 @@ internal static partial class HttpLoggingExtensions
     [LoggerMessage(5, LogLevel.Debug, "Decode failure while converting body.", EventName = "DecodeFailure")]
     public static partial void DecodeFailure(this ILogger logger, Exception ex);
 
-    [LoggerMessage(6, LogLevel.Debug, "Unrecognized Content-Type for body.", EventName = "UnrecognizedMediaType")]
-    public static partial void UnrecognizedMediaType(this ILogger logger);
+    [LoggerMessage(6, LogLevel.Debug, "Unrecognized Content-Type for request body.", EventName = "RequestUnrecognizedMediaType")]
```

Comment:
Let's keep the same id for both logs to avoid any possible breakage no matter how unlikely. We should be able to parameterize just the request/response part of the message.

I was going to suggest including *what* the unrecognized content-type was exactly. It seems useful for debugging, but then I realize most people logging the body are probably logging the header separately anyway.

Reply:
> I was going to suggest including _what_ the unrecognized content-type was exactly.

Should there be a different log when there _isn't_ a media type header?

Reply:
Is that a common scenario?

Reply:
Probably? Isn't that what the original issue was sort of about?

Reply:
Agreed. An "Unrecognized Content-Type" log when there isn't a "Content-Type" to begin with is super confusing. I would not log anything at all in that case.

Reply:
Added a new log message & a test